### PR TITLE
couple tweaks to optRemoveRedundantZeroInits

### PR DESCRIPTION
**Pull request by VSadov**
Make the optimizer for GC locals be aware that:

- CORINFO_HELP_INIT_PINVOKE_FRAME is not a GC safe point (not any more)
- there are other helpers that are not GC safe points

This removes some unnecessary initializations.

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=683630&view=ms.vss-build-web.run-extensions-tab)

### CODE REVIEW

**Comment by VSadov** [comment]
Example diff:

```diff
-15 (-7.54%) : 252648.dasm - (dynamicClass):IL_STUB_PInvoke(byref,byref,byref):int (FullOpts)
@@ -11,9 +11,9 @@
 ;  V01 arg1         [V01,T02] (  4,  4   )   byref  ->  rsi         single-def
 ;  V02 arg2         [V02,T03] (  4,  4   )   byref  ->  rdi         single-def
 ;* V03 loc0         [V03    ] (  0,  0   )     int  ->  zero-ref   
-;  V04 loc1         [V04    ] (  1,  1   )   byref  ->  [rbp-0x40]  must-init pinned single-def
-;  V05 loc2         [V05    ] (  1,  1   )   byref  ->  [rbp-0x48]  must-init pinned single-def
-;  V06 loc3         [V06    ] (  1,  1   )   byref  ->  [rbp-0x50]  must-init pinned single-def
+;  V04 loc1         [V04    ] (  1,  1   )   byref  ->  [rbp-0x40]  pinned single-def
+;  V05 loc2         [V05    ] (  1,  1   )   byref  ->  [rbp-0x48]  pinned single-def
+;  V06 loc3         [V06    ] (  1,  1   )   byref  ->  [rbp-0x50]  pinned single-def
 ;* V07 loc4         [V07    ] (  0,  0   )     int  ->  zero-ref   
 ;* V08 loc5         [V08    ] (  0,  0   )     int  ->  zero-ref   
 ;  V09 OutArgs      [V09    ] (  1,  1   )  struct (32) [rsp+0x00]  do-not-enreg[XS] addr-exposed "OutgoingArgSpace"
@@ -42,10 +42,6 @@ G_M13901_IG01:        ; bbWeight=1, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref,
        push     rbx
        sub      rsp, 136
        lea      rbp, [rsp+0xC0]
-       vxorps   xmm4, xmm4, xmm4
-       vmovdqa  xmmword ptr [rbp-0x50], xmm4
-       xor      eax, eax
-       mov      qword ptr [rbp-0x40], rax
        mov      rbx, rcx
        ; byrRegs +[rbx]
        mov      rsi, rdx
@@ -53,7 +49,7 @@ G_M13901_IG01:        ; bbWeight=1, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref,
        mov      rdi, r8
        ; byrRegs +[rdi]
        mov      r14, r10
-						;; size=54 bbWeight=1 PerfScore 13.33
+						;; size=39 bbWeight=1 PerfScore 9.75
 G_M13901_IG02:        ; bbWeight=1, gcrefRegs=0000 {}, byrefRegs=00C8 {rbx rsi rdi}, byref
        mov      qword ptr [rbp-0x60], r14
        lea      rcx, [rbp-0x98]
@@ -104,7 +100,7 @@ G_M13901_IG06:        ; bbWeight=1, epilog, nogc, extend
        ret      
 						;; size=20 bbWeight=1 PerfScore 5.25
 
-; Total bytes of code 199, prolog size 42, PerfScore 48.08, instruction count 56, allocated bytes for code 199 (MethodHash=7b4bc9b2) for method (dynamicClass):IL_STUB_PInvoke(byref,byref,byref):int (FullOpts)
+; Total bytes of code 184, prolog size 27, PerfScore 44.50, instruction count 52, allocated bytes for code 184 (MethodHash=7b4bc9b2) for method (dynamicClass):IL_STUB_PInvoke(byref,byref,byref):int (FullOpts)
```

**Review by EgorBo — APPROVED** [review]
LGTM!

File: src/coreclr/jit/optimizer.cpp [hunk 1]
```diff
@@ -5918,7 +5918,12 @@ void Compiler::optRemoveRedundantZeroInits()
             {
                 if (((tree->gtFlags & GTF_CALL) != 0))
                 {
-                    hasGCSafePoint = true;
+                    // if this is not a No-GC heleper
```
Comment (amanasifkhalid) [inline src/coreclr/jit/optimizer.cpp:5921]: Nit:
```suggestion
                    // if this is not a No-GC helper
```

**Comment by VSadov** [comment]
Thanks!

**Pull request merged**